src/cameras/search/hooks/use-input-props.js

```diff
@@ -1,22 +1,9 @@
 import { useSearchbarStyle } from 'cameras/search/hooks/use-searchbar-style'
-import { useContext } from 'cameras/context'
-import { useRequestSearch } from 'cameras/search/hooks/use-request-search'
-import { useCancelSearch } from 'cameras/search/hooks/use-cancel-search'
 
-const useCallbacks = () => {
-  const onChange = useRequestSearch()
-  const onCancelSearch = useCancelSearch()
-
-  return ({
-    onChange,
-    onCancelSearch,
-  })
-}
-
-export const useInputProps = () => (
-  {
-    value: useContext().searchPhrase,
-    placeholder: 'Cameras by id',
-    ...useCallbacks(),
-    classes: useSearchbarStyle(),
-  })
+export const useInputProps = setSearchInput => ({
+  onChange: value => {
```

Comment:
Since we need a function's reference passed to the `onChange` prop, we can avoid creating another lambda by passing in the reference to `setSearchInput`, like this:
![image](https://user-images.githubusercontent.com/23569938/129751192-52fe81aa-4893-402e-8381-35e7c47c278b.png)

Reply:
Done!